main/options/options.cc

```diff
@@ -1199,6 +1201,24 @@ void readOptions(Options &opts,
             }
         }
 
+        if (raw.count("packager-layer") && !opts.stripePackages) {
+            // Default values are available at raw[...], but don't show up in raw.count(...),
+            // so we can use this as a way to check if the user passed in the flag
+            logger->error("--packager-layer can only be specified in --stripe-packages mode");
+            throw EarlyReturnWithCode(1);
+        }
+
+        if (opts.stripePackages) {
+            std::regex layerValid("[a-zA-Z0-9]+");
```

Comment:
I picked the alphanumeric on a whim, open to feedback about being more or less restrictive here.

Reply:
Maybe put this in a comment?
What does the Stripe-internal implementation do?

Reply:
The regex in the stripe version is `[-A-Za-z0-9_/.]+`, but there’s also a fixed list of valid layers that are all alphabetic, so the regex is (sort of) irrelevant